src/Nancy/Responses/EmbeddedFileResponse.cs

```diff
@@ -47,19 +48,17 @@ private Stream GetResourceContent(Assembly assembly, string resourcePath, string
         {
             var resourceName = assembly
                 .GetManifestResourceNames()
-                .Where(x => GetFileNameFromResourceName(resourcePath, x).Equals(name, StringComparison.OrdinalIgnoreCase))
-                .Select(x => GetFileNameFromResourceName(resourcePath, x))
-                .FirstOrDefault();
+                .FirstOrDefault(x => GetFileNameFromResourceName(resourcePath, x).Equals(name, StringComparison.OrdinalIgnoreCase));

```

Comment:
Per-request performance concerns. 

Just a smell that needs to be thought about....